src/main/java/net/dv8tion/jda/internal/managers/ScheduledEventManagerImpl.java

```diff
@@ -215,11 +215,9 @@ private void preChecks()
     {
         if (shouldUpdate(LOCATION))
         {
-            Checks.check(getScheduledEvent().getStatus() == ScheduledEvent.Status.SCHEDULED, "Cannot update location of non-scheduled event.");
+            Checks.check(getScheduledEvent().getStatus() == ScheduledEvent.Status.SCHEDULED || (entityType == ScheduledEvent.Type.EXTERNAL && getScheduledEvent().getType() == ScheduledEvent.Type.EXTERNAL), "Cannot update location type or location channel of non-scheduled event.");
```

Comment:
Would it make more sense to just check if the type is equal to the current type?

```suggestion
            Checks.check(getScheduledEvent().getStatus() == ScheduledEvent.Status.SCHEDULED || entityType == getScheduledEvent().getType()), "Cannot update location type or location channel of non-scheduled event.");
```

Reply:
No, because that doesn't cover the type being Voice/Stage and the channel being changed, which is not possible. You can only edit the location value if the type is external.

Reply:
I just tested this and the client lets me change the channel for the event.

Reply:
Oh I see, you can't change it if the event is already taking place in that channel. That makes sense, then.